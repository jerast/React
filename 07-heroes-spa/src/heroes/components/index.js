export * from './HeroList';
export * from './HeroCard';
export * from './HeroPoster';
export * from './HeroNotFound';