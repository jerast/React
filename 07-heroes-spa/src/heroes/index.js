
// Components
export * from './components/HeroList';
export * from './components/HeroCard';
export * from './components/HeroPoster';
export * from './components/HeroNotFound';

// Data
export * from './data/heroes';

// Helpers
export * from './helpers/getHeroesById'
export * from './helpers/getHeroesByName'
export * from './helpers/getHeroesByPublisher'

// Pages
export * from './pages/MarvelPage';
export * from './pages/DcPage';
export * from './pages/SearchPage';
export * from './pages/HeroPage';

// Routes
export * from './routes/Heroes.Routes';