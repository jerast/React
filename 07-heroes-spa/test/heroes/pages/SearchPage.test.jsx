import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "@/heroes";

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
   useNavigate: () => mockedUseNavigate,
}));

describe('Testing on <SearchPage />', () => {

   beforeEach(() => jest.clearAllMocks());

   test('should match the snapshot', () => {
      const { container } = render(
         <MemoryRouter>
            <SearchPage />
         </MemoryRouter>
      );
      
      expect( container ).toMatchSnapshot();
   });

   test('should match the snapshot', () => {
      render(
         <MemoryRouter initialEntries={['/search?query=batman']}>
            <SearchPage />
         </MemoryRouter>
      );
      
      const input = screen.getByRole('textbox');
      expect( input.value ).toBe('batman');

      const img = screen.getAllByRole('img')[1];
      expect( img.src ).toContain('/images/heroes/dc-batman.jpg');
   });
   
   test('should display if hero is not found', () => {
      render(
         <MemoryRouter initialEntries={['/search?query=batman123']}>
            <SearchPage />
         </MemoryRouter>
      );
      
      const alertText = screen.getByRole('alert');
      expect( alertText.textContent ).toContain('No heros with [ batman123 ]');

      const alertImg = screen.getAllByRole('img')[1];
      expect( alertImg.src ).toContain('/images/icons/opps.png');
   });

   test('should Navigate when form is submited', () => {
      const searchValue = 'batman';
      render(
         <MemoryRouter>
            <SearchPage />
         </MemoryRouter>
      );

      const form = screen.getByRole( 'form' );
      const formInput = screen.getByRole( 'textbox' );

      fireEvent.input( formInput, { target: { value: searchValue } } );
      fireEvent.submit( form );

      expect( mockedUseNavigate ).toHaveBeenCalledWith(`?query=${ searchValue }`);
   });

});