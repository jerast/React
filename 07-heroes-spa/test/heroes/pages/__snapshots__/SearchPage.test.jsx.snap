// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <SearchPage /> should match the snapshot 1`] = `
<div>
  <main
    class="grid p-8 place-items-center h-min"
  >
    <h1
      class="text-start animate__animated animate__fadeIn"
    >
      Search
    </h1>
    <form
      class="Search animate__animated animate__fadeIn"
      role="form"
    >
      <input
        class="Search__input"
        name="search"
        placeholder="Search a hero"
        type="text"
        value=""
      />
      <button
        class="Search__button"
      >
        <img
          alt="Search"
          src="https://cdn-icons-png.flaticon.com/16/622/622669.png"
        />
      </button>
    </form>
    <div
      class="Search__list animate__animated animate__fadeIn"
    />
  </main>
</div>
`;
